plugins/uefi-capsule/fu-plugin-uefi-capsule.c

```diff
@@ -366,6 +366,50 @@ fu_plugin_uefi_capsule_update_splash (FuPlugin *plugin, FuDevice *device, GError
 	return fu_plugin_uefi_capsule_write_splash_data (plugin, device, image_bmp, error);
 }
 
+static gboolean
+fu_capsule_update_on_disk (FuDevice *device, GError **error)
+{
+	g_autofree guint64 *os_indications_supported = NULL;
+	g_autofree guint64 *os_indications = NULL;
+	guint32 attr;
+	gsize os_ind_size = 0;
+
+	if (fu_device_get_metadata_boolean (device, "DisableCapsuleUpdateOnDisk"))
+		return FALSE;
+
+	/* check for capsuleupdate ondisk */
+	if (!fu_efivar_get_data (FU_EFIVAR_GUID_EFI_GLOBAL, "OsIndicationsSupported",
+				 (guint8 **) &os_indications_supported,
+				 &os_ind_size, &attr, error))
+		g_debug ("Can't Read OsIndicationsSupported");
+	if (*os_indications_supported & EFI_OS_INDICATIONS_FILE_CAPSULE_DELIVERY_SUPPORTED) {
+		if (!fu_efivar_get_data (FU_EFIVAR_GUID_EFI_GLOBAL, "OsIndications",
+				 (guint8 **) &os_indications,
+				 &os_ind_size, &attr, error)) {
+			*os_indications = 0;
+			g_debug ("Failed to read OsIndications");
+		}
+		//if u-boot
+		//cp capsule;
```

Comment:
where does it need to get set?

Reply:
You mean the u-boot detection part? Since it's only need to be tested once we can run a scan during coldplug?

Reply:
if you tell me how to do the commented out bits in meta-code (i.e. abstract) then I can fill in some of the fwupd bits.

Reply:
@hughsie, I think I sorted out the copy as well.  I'll update it and you can have a look

Reply:
Yes, but dont' stress. I was going to reorganise the plugin a bit to make it cleaner to add this support. The "detect" u-boot bit would be more useful tbh.

Reply:
@hughsie what @superm1 proposed actually solved all my questions. fu-uefi-device.c (which I completely missed) had all the infrastructure I needed.  So I've basically split the function into two parts as I suggested.  How does this work now? Do I push an update or force push it?